tests/services/logging.service.spec.ts

```diff
@@ -1,64 +1,128 @@
+      const expectedLog = `${headerLog}${sessionSeparator}${headerLog}`;
+      expect(actualLog).toEqual(expectedLog);
     });
 
-    it('should return updated log if log in cache contains max number of sessions', () => {
-      // Number of logs must stay the same
-      expect(getFilteredLogCount(repeatOldLogStartHeader(loggingService.LOG_COUNT_LIMIT), logHeaderFilter)).toEqual(
-        loggingService.LOG_COUNT_LIMIT
-      );
+    it('should successfully reinitialize logging session when limit reached', () => {
+      Array(loggingService.LOG_COUNT_LIMIT)
+        .fill(0)
+        .forEach(() => {
+          loggingService.startSession();
+          loggingService.reset();
+        });
+      loggingService.startSession();
+      const actualLog = loggingService.getCachedLog();
+      const expectedLog = Array(loggingService.LOG_COUNT_LIMIT)
+        .fill('')
+        .map((_) => headerLog)
+        .join(sessionSeparator);
+      expect(actualLog).toEqual(expectedLog);
+    });
+  });
 
-      // Number of Old Logs must be reduced by 1 (To make way for the new session log)
-      expect(getFilteredLogCount(repeatOldLogStartHeader(loggingService.LOG_COUNT_LIMIT), oldLogFilter)).toEqual(
-        loggingService.LOG_COUNT_LIMIT - 1
-      );
+  describe('.reset()', () => {
+    it('should do nothing if no session is ongoing', () => {
+      loggingService.reset();
+      const actualLog = loggingService.getCachedLog();
+      expect(actualLog).toBeNull();
     });
 
-    it('should return trimmed and updated log if log in cache exceeds max number of sessions', () => {
-      const exceededSessionCount = loggingService.LOG_COUNT_LIMIT + 10; // Arbitrary Exceed Count
+    it('should not tamper with existing log histories', () => {
+      let expectedLog = headerLog;
+      for (let i = 0; i < loggingService.LOG_COUNT_LIMIT + 1; i += 1) {
+        loggingService.startSession();
+        loggingService.reset();
+        const actualLog = loggingService.getCachedLog();
+        expect(actualLog).toEqual(expectedLog);
+        if (i < loggingService.LOG_COUNT_LIMIT - 1) {
+          expectedLog += `${sessionSeparator}${headerLog}`;
+        }
+      }
+    });
+  });
 
-      // Number of logs must be at max number
-      expect(getFilteredLogCount(repeatOldLogStartHeader(exceededSessionCount), logHeaderFilter)).toEqual(loggingService.LOG_COUNT_LIMIT);
+  describe('adding logs', () => {
+    it('should successfully add info logs', () => {
+      testAddLog(infoLogMessage);
+    });
+  });
 
-      // Number of Old Logs must be Max - 1 (To make way for the new session log)
-      expect(getFilteredLogCount(repeatOldLogStartHeader(exceededSessionCount), oldLogFilter)).toEqual(loggingService.LOG_COUNT_LIMIT - 1);
+  describe('updating and trimming logs from sessions', () => {
+    it('should trim oldest log if number of sessions exceed session limit', () => {
+      for (let i = 0; i < loggingService.LOG_COUNT_LIMIT + 1; i += 1) {
```

Comment:
you may use a forEach loop here

Reply:
Oh yeah I missed this, thanks for the suggestion!